pkg/authenticator/config/configuration_factory.go

```diff
@@ -44,6 +45,8 @@ func ConfigFromEnv(readFileFunc common.ReadFileFunc) (Configuration, error) {
 func getConfiguration(url string) (Configuration, error) {
 	if strings.Contains(url, k8sAuthenticator.AuthnType) {
 		return &k8sAuthenticator.Config{}, nil
+	} else if strings.Contains(url, jwtAuthenticator.AuthnType) {
```

Comment:
Instead of using an `if/else` block here, you could optionally use `switch` without a statement (i.e. evaluating expressions in the case statements). I think this makes it a little more obvious that there are 3 possible results for which we're looking. For example:
```
	switch {
	case strings.Contains(url, k8sAuthenticator.AuthnType):
		return &k8sAuthenticator.Config{}, nil
	case strings.Contains(url, jwtAuthenticator.AuthnType):
		return &jwtAuthenticator.Config{}, nil
	default:
		return nil, fmt.Errorf(log.CAKC063, url)
	}
```